### ISSUE TITLE
Bqplot doesn't display anything in jupyter notebook

### ISSUE THREAD

**Issue by dusadvarshit** [bloomberg/bqplot#1042]
**Describe the bug**
Bqplot does not display anything in the jupyter notebook. My issue is identical to that of https://github.com/bloomberg/bqplot/issues/677 but I do not follow the solution provided there.
I have not 

**To Reproduce**
import numpy as np
import bqplot.pyplot as bqplt

size = 100
scale = 100
np.random.seed(0)
x_data = np.arange(size)
y_data = np.cumsum(np.random.randn(size)  * scale)

fig = bqplt.figure(title='First Example')
bqplt.plot(x_data, y_data)
fig

**Expected behavior**
Display a line chart as demonstrated in the Line Chart section of Basic Plotting tutorial following the link:
https://github.com/bloomberg/bqplot/blob/master/examples/Basic%20Plotting/Basic%20Plotting.ipynb

**Screenshots**
![image](https://user-images.githubusercontent.com/7974289/75203266-b5cad780-5793-11ea-9270-57c1fa0886c0.png)

**Environment (please complete the following information):**
 - Operating System and Version: Ubuntu Linux 16.04.6 LTS {server, not desktop}
 - Browser [e.g. Firefox, Safari] (if applicable): Chrome & Firefox

**Additional context**
I just downloaded bqplot (via conda) today and was giving the test run in a new conda environment.
Also, ipywidgets seem to work well.
![image](https://user-images.githubusercontent.com/7974289/75203612-97191080-5794-11ea-9bd4-7182a93654e2.png)

**Comment by jasongrout** [comment]
Please include the bqplot version, and any errors from the browser Javascript console.

**Comment by dusadvarshit** [comment]
Thank you, Jason, for your comment. My bqplot version is 0.12.3

I am unsure how to provide errors from the browser Javascript console. I have never done that before. Could you please provide me some insight into the process?

**Comment by jasongrout** [comment]
You can open the javascript console in Chrome with the menu View > Developer > Javascript console. In Firefox, it is Tools > Web Developer > Web Console.

**Comment by dusadvarshit** [comment]
Hi Jason, 
I followed your steps and inspected the following 3 lines of codes in my chrome
fig = plt.figure(title='First Example')
plt.plot(y_data)
fig

The javascript console showed 38 errors and 1 warning.  I am copying all of them below.

------------- Warning and error below this line-------------------------

main.min.js?v=1bc4ae6ca8730320ff059b16de698691:47073 actions jupyter-notebook:find-and-replace does not exist, still binding it in case it will be defined later...
MenuBar.bind_events @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:47073
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:47981 Loaded moment locale en
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:24086 load_extensions Arguments(2)
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:24063 Loading extension: plotlywidget/extension
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:24063 Loading extension: jupyter-js-widgets/extension
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:38291 Session: kernel_created (d25bcd32-23d5-4231-ab12-e9930101adf3)
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37449 Starting WebSockets: ws://localhost:8000/api/kernels/3ddbaac4-6196-4a17-8ed6-b7cc7ea502a5
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37093 Kernel: kernel_connected (3ddbaac4-6196-4a17-8ed6-b7cc7ea502a5)
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37093 Kernel: kernel_ready (3ddbaac4-6196-4a17-8ed6-b7cc7ea502a5)
Widget_development.ipynb:1 Access to font at 'about:blank' from origin 'http://localhost:8000' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https.
about:blank:1 Failed to load resource: net::ERR_FAILED
bqplot.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
2main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
2main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
2require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)

**Comment by glyfnet** [comment]
same issue using SageMaker notebook with Python3. Is there a resolution yet?

**Comment by jasongrout** [comment]
SageMaker may not support bqplot and other widgets - please contact their support about that.

**Comment by jasongrout** [comment]
@dusadvarshit - I would try installing into a clean new environment.

**Comment by dusadvarshit** [comment]
@jasongrout I tried installing into a new environment and still no plotting is happening. Here is the error log from the browser
--------------------------------------------------------------------------------------------------------------
Failed to load resource: the server responded with a status of 404 (Not Found)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
2main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
manager-base.js:273 Could not instantiate widget
(anonymous) @ manager-base.js:273
utils.js:119 Error: Could not create a model.
    at utils.js:119
(anonymous) @ utils.js:119
2main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995 Couldn't process kernel message Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)
(anonymous) @ main.min.js?v=1bc4ae6ca8730320ff059b16de698691:37995
2require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168 Uncaught (in promise) Error: Script error for "bqplot"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:168)
    at HTMLScriptElement.onScriptError (require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1735)

**Comment by jasongrout** [comment]
@dusadvarshit - can you paste your installation log? Were there any errors in installation? Can you give step-by-step instructions for recreating your clean environment so we can reproduce your setup?

**Comment by dusadvarshit** [comment]
No @jasongrout I did not face any errors during installation. 
Here is the process I followed for my installation.

 conda create -n bqplot python=3.6 
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/rohit/sdd/miniconda3/envs/bqplot

  added / updated specs:
    - python=3.6


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2020.1.1   |                0         125 KB
    certifi-2019.11.28         |           py36_0         153 KB
    ------------------------------------------------------------
                                           Total:         278 KB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.1.1-0
  certifi            pkgs/main/linux-64::certifi-2019.11.28-py36_0
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
  libedit            pkgs/main/linux-64::libedit-3.1.20181209-hc058e9b_0
  libffi             pkgs/main/linux-64::libffi-3.2.1-hd88cf55_4
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_0
  openssl            pkgs/main/linux-64::openssl-1.1.1d-h7b6447c_4
  pip                pkgs/main/linux-64::pip-20.0.2-py36_1
  python             pkgs/main/linux-64::python-3.6.10-h0371630_0
  readline           pkgs/main/linux-64::readline-7.0-h7b6447c_5
  setuptools         pkgs/main/linux-64::setuptools-46.0.0-py36_0
  sqlite             pkgs/main/linux-64::sqlite-3.31.1-h7b6447c_0
  tk                 pkgs/main/linux-64::tk-8.6.8-hbc83047_0
  wheel              pkgs/main/linux-64::wheel-0.34.2-py36_0
  xz                 pkgs/main/linux-64::xz-5.2.4-h14c3975_4
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3


Proceed ([y]/n)? y


Downloading and Extracting Packages
ca-certificates-2020 | 125 KB    | ####################################################################################################################################################################### | 100%
certifi-2019.11.28   | 153 KB    | ####################################################################################################################################################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate bqplot
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(pdf_read) rohit@livscience:~$ conda activate bqplot
(bqplot) rohit@livscience:~$ conda install -c anaconda jupyter
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/rohit/sdd/miniconda3/envs/bqplot

  added / updated specs:
    - jupyter


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2020.1.1   |                0         132 KB  anaconda
    certifi-2019.11.28         |           py36_0         156 KB  anaconda
    six-1.14.0                 |           py36_0          27 KB  anaconda
    ------------------------------------------------------------
                                           Total:         316 KB

The following NEW packages will be INSTALLED:

  attrs              anaconda/noarch::attrs-19.3.0-py_0
  backcall           anaconda/linux-64::backcall-0.1.0-py36_0
  bleach             anaconda/linux-64::bleach-3.1.0-py36_0
  dbus               anaconda/linux-64::dbus-1.13.12-h746ee38_0
  decorator          anaconda/noarch::decorator-4.4.1-py_0
  defusedxml         anaconda/noarch::defusedxml-0.6.0-py_0
  entrypoints        anaconda/linux-64::entrypoints-0.3-py36_0
  expat              anaconda/linux-64::expat-2.2.6-he6710b0_0
  fontconfig         anaconda/linux-64::fontconfig-2.13.0-h9420a91_0
  freetype           anaconda/linux-64::freetype-2.9.1-h8a8886c_1
  glib               anaconda/linux-64::glib-2.56.2-hd408876_0
  gmp                anaconda/linux-64::gmp-6.1.2-hb3b607b_0
  gst-plugins-base   anaconda/linux-64::gst-plugins-base-1.14.0-hbbd80ab_1
  gstreamer          anaconda/linux-64::gstreamer-1.14.0-hb453b48_1
  icu                anaconda/linux-64::icu-58.2-h211956c_0
  importlib_metadata anaconda/linux-64::importlib_metadata-1.5.0-py36_0
  ipykernel          anaconda/linux-64::ipykernel-5.1.4-py36h39e3cac_0
  ipython            anaconda/linux-64::ipython-7.12.0-py36h5ca1d4c_0
  ipython_genutils   anaconda/linux-64::ipython_genutils-0.2.0-py36_0
  ipywidgets         anaconda/noarch::ipywidgets-7.5.1-py_0
  jedi               anaconda/linux-64::jedi-0.16.0-py36_0
  jinja2             anaconda/noarch::jinja2-2.11.1-py_0
  jpeg               anaconda/linux-64::jpeg-9b-habf39ab_1
  jsonschema         anaconda/linux-64::jsonschema-3.2.0-py36_0
  jupyter            anaconda/linux-64::jupyter-1.0.0-py36_7
  jupyter_client     anaconda/linux-64::jupyter_client-5.3.4-py36_0
  jupyter_console    anaconda/noarch::jupyter_console-6.1.0-py_0
  jupyter_core       anaconda/linux-64::jupyter_core-4.6.1-py36_0
  libpng             anaconda/linux-64::libpng-1.6.37-hbc83047_0
  libsodium          anaconda/linux-64::libsodium-1.0.16-h1bed415_0
  libuuid            anaconda/linux-64::libuuid-1.0.3-h1bed415_2
  libxcb             anaconda/linux-64::libxcb-1.13-h1bed415_1
  libxml2            anaconda/linux-64::libxml2-2.9.9-hea5a465_1
  markupsafe         anaconda/linux-64::markupsafe-1.1.1-py36h7b6447c_0
  mistune            anaconda/linux-64::mistune-0.8.4-py36h7b6447c_0
  nbconvert          anaconda/linux-64::nbconvert-5.6.1-py36_0
  nbformat           anaconda/noarch::nbformat-5.0.4-py_0
  notebook           anaconda/linux-64::notebook-6.0.3-py36_0
  pandoc             anaconda/linux-64::pandoc-2.2.3.2-0
  pandocfilters      anaconda/linux-64::pandocfilters-1.4.2-py36_1
  parso              anaconda/noarch::parso-0.6.1-py_0
  pcre               anaconda/linux-64::pcre-8.43-he6710b0_0
  pexpect            anaconda/linux-64::pexpect-4.8.0-py36_0
  pickleshare        anaconda/linux-64::pickleshare-0.7.5-py36_0
  prometheus_client  anaconda/noarch::prometheus_client-0.7.1-py_0
  prompt_toolkit     anaconda/noarch::prompt_toolkit-3.0.3-py_0
  ptyprocess         anaconda/linux-64::ptyprocess-0.6.0-py36_0
  pygments           anaconda/noarch::pygments-2.5.2-py_0
  pyqt               anaconda/linux-64::pyqt-5.9.2-py36h22d08a2_1
  pyrsistent         anaconda/linux-64::pyrsistent-0.15.7-py36h7b6447c_0
  python-dateutil    anaconda/noarch::python-dateutil-2.8.1-py_0
  pyzmq              anaconda/linux-64::pyzmq-18.1.1-py36he6710b0_0
  qt                 anaconda/linux-64::qt-5.9.7-h5867ecd_1
  qtconsole          anaconda/noarch::qtconsole-4.6.0-py_1
  send2trash         anaconda/linux-64::send2trash-1.5.0-py36_0
  sip                anaconda/linux-64::sip-4.19.13-py36he6710b0_0
  six                anaconda/linux-64::six-1.14.0-py36_0
  terminado          anaconda/linux-64::terminado-0.8.3-py36_0
  testpath           anaconda/noarch::testpath-0.4.4-py_0
  tornado            anaconda/linux-64::tornado-6.0.3-py36h7b6447c_3
  traitlets          anaconda/linux-64::traitlets-4.3.3-py36_0
  wcwidth            anaconda/noarch::wcwidth-0.1.8-py_0
  webencodings       anaconda/linux-64::webencodings-0.5.1-py36_1
  widgetsnbextension anaconda/linux-64::widgetsnbextension-3.5.1-py36_0
  zeromq             anaconda/linux-64::zeromq-4.3.1-he6710b0_3
  zipp               anaconda/noarch::zipp-2.2.0-py_0

The following packages will be UPDATED:

  openssl              pkgs/main::openssl-1.1.1d-h7b6447c_4 --> anaconda::openssl-1.1.1-h7b6447c_0

The following packages will be SUPERSEDED by a higher-priority channel:

  ca-certificates                                 pkgs/main --> anaconda
  certifi                                         pkgs/main --> anaconda


Proceed ([y]/n)? y


Downloading and Extracting Packages
six-1.14.0           | 27 KB     | ####################################################################################################################################################################### | 100%
certifi-2019.11.28   | 156 KB    | ####################################################################################################################################################################### | 100%
ca-certificates-2020 | 132 KB    | ####################################################################################################################################################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(bqplot) rohit@livscience:~$ conda install -c conda-forge bqplot
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/rohit/sdd/miniconda3/envs/bqplot

  added / updated specs:
    - bqplot


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.11.28         |   py36h9f0ad1d_1         149 KB  conda-forge
    libblas-3.8.0              |      14_openblas          10 KB  conda-forge
    libcblas-3.8.0             |      14_openblas          10 KB  conda-forge
    libgfortran-ng-7.3.0       |       hdf63c60_5         1.7 MB  conda-forge
    liblapack-3.8.0            |      14_openblas          10 KB  conda-forge
    libopenblas-0.3.7          |       h5ec1e0e_6         7.6 MB  conda-forge
    numpy-1.18.1               |   py36h95a1406_0         5.2 MB  conda-forge
    pandas-1.0.1               |   py36hb3f55d8_0        11.1 MB  conda-forge
    python_abi-3.6             |          1_cp36m           4 KB  conda-forge
    pytz-2019.3                |             py_0         237 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        26.0 MB

The following NEW packages will be INSTALLED:

  bqplot             conda-forge/noarch::bqplot-0.12.5-py_0
  libblas            conda-forge/linux-64::libblas-3.8.0-14_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-14_openblas
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.3.0-hdf63c60_5
  liblapack          conda-forge/linux-64::liblapack-3.8.0-14_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.7-h5ec1e0e_6
  numpy              conda-forge/linux-64::numpy-1.18.1-py36h95a1406_0
  pandas             conda-forge/linux-64::pandas-1.0.1-py36hb3f55d8_0
  python_abi         conda-forge/linux-64::python_abi-3.6-1_cp36m
  pytz               conda-forge/noarch::pytz-2019.3-py_0
  traittypes         conda-forge/noarch::traittypes-0.2.1-py_1

The following packages will be UPDATED:

  certifi               anaconda::certifi-2019.11.28-py36_0 --> conda-forge::certifi-2019.11.28-py36h9f0ad1d_1

The following packages will be SUPERSEDED by a higher-priority channel:

  ca-certificates      anaconda::ca-certificates-2020.1.1-0 --> conda-forge::ca-certificates-2019.11.28-hecc5488_0
  openssl                anaconda::openssl-1.1.1-h7b6447c_0 --> conda-forge::openssl-1.1.1d-h516909a_0


Proceed ([y]/n)? y


Downloading and Extracting Packages
libcblas-3.8.0       | 10 KB     | ####################################################################################################################################################################### | 100%
pandas-1.0.1         | 11.1 MB   | ####################################################################################################################################################################### | 100%
liblapack-3.8.0      | 10 KB     | ####################################################################################################################################################################### | 100%
numpy-1.18.1         | 5.2 MB    | ####################################################################################################################################################################### | 100%
pytz-2019.3          | 237 KB    | ####################################################################################################################################################################### | 100%
libgfortran-ng-7.3.0 | 1.7 MB    | ####################################################################################################################################################################### | 100%
certifi-2019.11.28   | 149 KB    | ####################################################################################################################################################################### | 100%
libblas-3.8.0        | 10 KB     | ####################################################################################################################################################################### | 100%
python_abi-3.6       | 4 KB      | ####################################################################################################################################################################### | 100%
libopenblas-0.3.7    | 7.6 MB    | ####################################################################################################################################################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done

**After installation steps, I attempted the bqplot tutorial available on your github page.**
import numpy as np
import bqplot.pyplot as bqplt

size = 100
scale = 100
np.random.seed(0)
x_data = np.arange(size)
y_data = np.cumsum(np.random.randn(size) * scale)

fig = bqplt.figure(title='First Example')
bqplt.plot(x_data, y_data)
fig

**Nothing happens in the end.**

**Comment by jasongrout** [comment]
Okay, thanks, your installation looks good. I just tried it as well with Then are you launching the notebook with `jupyter notebook` with that environment activated?

What is the output from this on the command line?

```
jupyter nbextension list
```

And just to make sure, what is the output of this in the notebook:
```
import bqplot
print(bqplot.__file__)
print(bqplot.__version__)
import ipywidgets
print(ipywidgets.__file__)
print(ipywidgets.__version__)
```
The relevant error from your javascript log above is:

>  bqplot.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)

Can you start the notebook with `jupyter notebook --debug`, and look for a lines like:

```
[D 12:10:51.149 NotebookApp] Path bqplot/extension.js served from ...
[D 12:10:51.151 NotebookApp] 200 GET /nbextensions/bqplot/extension.js...
```
and paste them here?

**Comment by dusadvarshit** [comment]
Hi Jason,

Following are the responses to all the questions you asked:

`Output for jupyter nbextension list:

(bqplot) rohit@livscience:~$ jupyter nbextension list
Known nbextensions:
  config dir: /home/rohit/sdd/miniconda3/envs/bqplot/etc/jupyter/nbconfig
    notebook section
      bqplot/extension  enabled
      - Validating: OK
      jupyter-js-widgets/extension  enabled
      - Validating: OK
  config dir: /usr/local/etc/jupyter/nbconfig
    notebook section
      plotlywidget/extension  enabled
      - Validating: OK
      jupyter-js-widgets/extension  enabled
      - Validating: OK
(bqplot) rohit@livscience:~$`


`Output for 
import bqplot
print(bqplot.__file__)
print(bqplot.__version__)
import ipywidgets
print(ipywidgets.__file__)
print(ipywidgets.__version__)`

```

**Comment by burizao** [comment]
I had the same problem but when I ran on debug mode, using a command line, it worked. Then I tried again without the "--debug" and worked too. However, If I ran through the Jupyter link , the figure doesn´t show.
